### PR TITLE
extract interfaces and data models for validation

### PR DESCRIPTION
**Pull request by 928PJY**

### CODE REVIEW

File: All.sln [hunk 1]
```diff
@@ -220,6 +220,8 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "MarkdownFragments", "Markdo
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "YamlSplitter", "tools\YamlSplitter\YamlSplitter.csproj", "{2CA09972-F737-459A-A79E-EB96CC726200}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.DocAsCode.MarkdigEngine.Validator", "src\Microsoft.DocAsCode.MarkdigEngine.Validator\Microsoft.DocAsCode.MarkdigEngine.Validator.csproj", "{BD35FD90-4F00-4A19-9530-2279A6A15EE8}"
```
Comment (vicancy) [inline All.sln]: naming: use plural `Validators`?

File: src/Microsoft.DocAsCode.MarkdigEngine.Extensions/Validation/MarkdownValidatorBuilder.cs [hunk 2]
```diff
@@ -11,6 +11,7 @@ namespace Microsoft.DocAsCode.MarkdigEngine.Extensions
     using Markdig.Syntax;
     using Microsoft.DocAsCode.Common;
     using Microsoft.DocAsCode.Plugins;
+    using Microsoft.DocAsCode.MarkdigEngine.Validator;
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.MarkdigEngine.Extensions/Validation/MarkdownValidatorBuilder.cs]: order

File: test/Microsoft.DocAsCode.MarkdigEngine.Tests/HtmlMarkdownObjectValidatorProvider.cs [hunk 3]
```diff
@@ -6,11 +6,10 @@ namespace Microsoft.DocAsCode.MarkdigEngine.Tests
...
     using System.Composition;
 
-    using MarkdigEngine.Extensions;
-
     using Markdig.Syntax;
     using Microsoft.DocAsCode.Common;
-
+    using Microsoft.DocAsCode.MarkdigEngine.Validators;
+    
     [Export(ContractName, typeof(IMarkdownObjectValidatorProvider))]
```
Comment (seshenoy) [inline test/Microsoft.DocAsCode.MarkdigEngine.Tests/HtmlMarkdownObjectValidatorProvider.cs]: Does the test require to have the Export attribute based on the new design we talked ?

**Comment by seshenoy** [comment]
Looks good...

**Pull request merged**